src/marks/raster.js

```diff
@@ -95,10 +95,11 @@ export class Raster extends AbstractRaster {
   }
   // Ignore the color scale, so the fill channel is returned unscaled.
   scale(channels, {color, ...scales}, context) {
+    this.recolor = channels.fill?.scale;
     return super.scale(channels, scales, context);
   }
   render(index, scales, channels, dimensions, context) {
-    const color = scales.color ?? ((x) => x);
+    const color = this.recolor && scales.color ? scales.color : (x) => x;
```

Comment:
I think you want something like this:

```suggestion
    const color = scales[this.channels.fill?.scale] ?? (x) => x;
```

Reply:
hadn't thought about that, but it makes sense!

Reply:
this.channels.fill.scale comes in as "auto" and not "color", so I had to adapt a bit